src/Signup.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import axios from "axios";
+import history from './history';
+import './todo.css';
+
+class Signup extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            username: "",
+            password: "",
+            confirmPassword: "",
+            cred: {}
+        };
+    }
+
+    setUsername = event => {
+        let value = event.target.value;
```

Comment:
any specific reason assigning  event.target.value to variable "value" then setting to state?

Reply:
Yes it is complex, I have to simplify that